pkg/stanza/fileconsumer/trie.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TRIE data structure inspired by https://github.com/dghubble/trie
+// This defers from the original trie.
+
+package fileconsumer
+
+type Trie struct {
+	value    interface{}
+	children map[byte]*Trie
+}
+
+// Trie node and the part string key of the child the path descends into.
+type nodeTrie struct {
+	node *Trie
+	b    byte
+}
+
+// NewPathTrie allocates and returns a new *Trie.
+func NewTrie() *Trie {
+	return &Trie{}
+}
+
+func (trie *Trie) Get(key []byte) interface{} {
+	node := trie
+	for _, r := range key {
+		// We have reached end of the current path and all the previous characters have matched
+		// Return if current node is leaf and it is not root
+		if node.isLeaf() && node != trie {
+			return r
```

Comment:
Isn't `r` just the last byte?
```suggestion
			return key
```